diff --git a/hw/arm/Kconfig b/hw/arm/Kconfig
index 6f5b2ad..8eff4e1 100644
--- a/hw/arm/Kconfig
+++ b/hw/arm/Kconfig
@@ -727,3 +727,10 @@ config ARMSSE
     select UNIMP
     select SSE_COUNTER
     select SSE_TIMER
+
+config RZA1H
+    bool
+    default y
+    depends on TCG && ARM
+    select ARM_GIC
+    select UNIMP
diff --git a/hw/arm/meson.build b/hw/arm/meson.build
index 4233a80..05ad4cd 100644
--- a/hw/arm/meson.build
+++ b/hw/arm/meson.build
@@ -111,5 +111,6 @@ arm_common_ss.add(when: 'CONFIG_VERSATILE', if_true: files('versatilepb.c'))
 arm_common_ss.add(when: 'CONFIG_VEXPRESS', if_true: files('vexpress.c'))
 
 arm_common_ss.add(files('boot.c'))
+arm_common_ss.add(when: 'CONFIG_RZA1H', if_true: files('rz_a1h.c', 'ostm.c', 'spi_boot.c', 'gpio.c', 'l2c.c', 'scif.c'))
 
 hw_common_arch += {'arm': arm_common_ss}
